### PR TITLE
CIV-8154 flight delay

### PR DESCRIPTION
**Pull request by GorilaNaranja**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-8154

### Change description ###
CUI (Money Claims): Additional Screen for Flight Delay

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/modules/i18n/locales/cy.json [hunk 1]
```diff
@@ -1761,6 +1761,15 @@
       "PAGE_TITLE": "Tystiolaeth y diffynnydd ei hun",
       "TITLE": "A ydych eisiau rhoi tystiolaeth eich hun?"
     },
+    "DELAYED_FLIGHT": {
+      "CLAIMING_FOR_DELAYED": "A ydych yn hawlio am oedi wrth hedfan?"
+    },
+    "FLIGHT_DETAILS": {
+      "FLIGHT_DETAILS": "Rhowch fanylion y daith hedfan",
+      "AIRLINE": "Cwmni hedfan",
```
Comment (manish14836) [inline src/main/modules/i18n/locales/cy.json:1769]: Sort these for both files cy.json and en.json

File: src/main/modules/i18n/locales/cy.json [hunk 2]
```diff
@@ -3656,6 +3665,15 @@
     "DESCRIPTION_REQUIRED": "Rhowch ddisgrifiad o beth ddigwyddodd",
     "DETAILS_REQUIRED": "Rhowch fanylion",
     "DIRECTION_QUESTIONNAIRE_REQUIRED_MESSAGE": "Dywedwch wrthym os ydych yn credu bod manylion yr anghenion ar gyfer y gwrandawiad ar y dudalen hon yn wir",
+    "DELAYED_FLIGHT": {
+      "CLAIMING_FOR_DELAY_REQUIRED": "Mae angen dewis opsiwn ar gyfer A ydych yn hawlio am oedi wrth hedfan"
+    },
+    "FLIGHT_DETAILS": {
+      "AIRLINE_REQUIRED": "Mae angen manylion y cwmni hedfan",
+      "FLIGHT_NUMBER_REQUIRED": "Mae angen rhif hedfan",
```
Comment (manish14836) [inline src/main/modules/i18n/locales/cy.json:3673]: Sort these for both files cy.json and en.json

File: src/main/routes/features/claim/airlines/flightDetailsController.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    }
+  } catch (error) {
+    next(error);
+  }
+}) as RequestHandler);
+
+const buildDataList = (airlines: AirlineList[] = [], hasAirlineError: boolean, selection = '', lng: string) => {
+  let options = '';
+  airlines = airlines.filter(item => item.airline !== 'OTHER');
+  airlines.forEach((airline) => {
```
Comment (manish14836) [inline src/main/routes/features/claim/airlines/flightDetailsController.ts:56]: you can chain them for more readability

File: src/main/views/features/claim/airlines/delayed-flight.njk [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{% extends "claim-details-tpl.njk" %}
```
[the remark this answers is not in the document]
Comment (GorilaNaranja) [reply src/main/views/features/claim/airlines/delayed-flight.njk:1]: This ticket is for create claim journey (/claim)

File: src/main/routes/features/claim/airlines/flightDetailsController.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+const flightDetailsController = Router();
+const flightDetailsPath = 'features/claim/airlines/flight-details';
+const civilServiceApiBaseUrl = config.get<string>('services.civilService.url');
+const civilServiceClient: CivilServiceClient = new CivilServiceClient(civilServiceApiBaseUrl);
+
+flightDetailsController.get(FLIGHT_DETAILS_URL, (async (req: AppRequest, res: Response, next: NextFunction) => {
+  try {
+    const lng = req.query.lang ? req.query.lang : req.cookies.lang;
+    const claimId = req.session?.user?.id;
+    const flightDetails = await getFlightDetails(claimId);
```
[the remark this answers is not in the document]
Comment (GorilaNaranja) [reply src/main/routes/features/claim/airlines/flightDetailsController.ts:21]: Under claim journey we don't have claim id yet.

File: src/main/routes/features/claim/airlines/flightDetailsController.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+const buildDataList = (airlines: AirlineList[] = [], hasAirlineError: boolean, selection = '', lng: string) => {
+  let options = '';
+  airlines = airlines.filter(item => item.airline !== 'OTHER');
+  airlines.forEach((airline) => {
+    options += `<option value="${airline.airline}"> `;
+  });
+
+  return `
+    <div class="${hasAirlineError ? 'govuk-form-group--error govuk-!-margin-bottom-6' : 'govuk-form-group'}">
```
[the remark this answers is not in the document]
Comment (GorilaNaranja) [reply src/main/routes/features/claim/airlines/flightDetailsController.ts:61]: It has to be in a ts file

File: src/test/unit/routes/features/claim/airlines/delayedFlightController.test.ts [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+} from 'routes/urls';
+import {TestMessages} from '../../../../../utils/errorMessageTestConstants';
+import {t} from 'i18next';
+import {Claim} from 'models/claim';
+import {mockCivilClaim} from '../../../../../utils/mockDraftStore';
+import {YesNo} from 'common/form/models/yesNo';
+import * as draftStoreService from 'modules/draft-store/draftStoreService';
+
+jest.mock('../../../../../../main/modules/oidc');
+jest.mock('../../../../../../main/modules/draft-store');
```
[the remark this answers is not in the document]
Comment (GorilaNaranja) [reply src/test/unit/routes/features/claim/airlines/delayedFlightController.test.ts:18]: removed

**Review by laarmada — APPROVED** [review]

File: src/main/modules/i18n/locales/cy.json (hunk 2 above)
Comment (GorilaNaranja) [reply src/main/modules/i18n/locales/cy.json]: fixed

**Review by laarmada — APPROVED** [review]

**Review by manish14836 — APPROVED** [review]

**Pull request merged**